xray/fasthttp.go

```diff
@@ -0,0 +1,92 @@
+package xray
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+
+	"github.com/aws/aws-xray-sdk-go/header"
+	"github.com/valyala/fasthttp"
+)
+
+// FastHTTPHandler
+type FastHTTPHandler struct {
```

Comment:
Can we make this struct private? don't see any use of making this public.

Reply:
Okay, I made that change. But I also created a `FastHTTPHandler` interface and here's a use case (you can also see it in the readme): 

```go
func middleware(name string, h fasthttp.RequestHandler, fh xray.FastHTTPHandler) fasthttp.RequestHandler {
	f := func(ctx *fasthttp.RequestCtx) {
		h(ctx)
	}

	return fh.Handler(xray.NewFixedSegmentNamer(name), f)
}

func main() {
	fh := xray.NewFastHTTPInstrumentor(nil)
	r := router.New()
	r.GET("/", middleware("index", index, fh))
	r.GET("/hello/{name}", middleware("hello", hello, fh))

	log.Fatal(fasthttp.ListenAndServe(":8080", r.Handler))
}

```